### PR TITLE
TST: Inconsistent behavior of .replace() in Int64 series with NA 

### LINKED ISSUE #38267: BUG: Inconsistent behavior of `.replace()` in Int64 series with `<NA>`.
### Code Sample

This code does work as expected (0 value is replaced by `<NA>` in an Int64 series).
```python
>>> import pandas as pd
>>> s = pd.Series([0, 1]).astype("Int64")
>>> s.replace(0, pd.NA)
0    <NA>
1       1
dtype: Int64
```

In contrast, when having `<NA>` values in the series `.replace()` does not work as expected (0 is **not** replaced by `<NA>`).
```python
>>> import pandas as pd
>>> s = pd.Series([0, None]).astype("Int64")
>>> s.replace(0, pd.NA)
0       0
1    <NA>
dtype: object
```

However, the code does work if we add `.fillna()` before replacing:
```python
>>> import pandas as pd
>>> s = pd.Series([0, None]).astype("Int64")
>>> s.fillna(0).replace(0, pd.NA)
0    <NA>
1    <NA>
dtype: Int64
```

### Problem description
When applying the replace on a Int64 series with or without `<NA>` I expect consistent behavior.
Also the result type of `.replace()` seems to be inconsistent based on the usage of `.fillna()` on the series (without `fillna()` the type changes to `object`).

### Expected behavior
I would expect that `s.replace(0, pd.NA)` and `s.fillna(0).replace(0, pd.NA)` on an Int64 series will **always** lead to the same result.

### Output of pd.show_versions()
<details>
  <summary>Details</summary>

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.8.0.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-53-generic
Version          : #59~18.04.1-Ubuntu SMP Wed Oct 21 12:14:56 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.4
numpy            : 1.19.2
pytz             : 2020.4
dateutil         : 2.8.1
pip              : 20.3
setuptools       : 50.3.2.post20201201
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None
</details>

### PR DESCRIPTION
**Pull request by ftrihardjo**
- [ ] closes #38267
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/methods/test_replace.py [hunk 1]
```diff
@@ -208,6 +208,14 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize('dtype', ['int8', 'int16', 'int32', 'int64'])
+    def test_replace_int_with_na(self, dtype):
+        result = pd.Series([0, None]).astype(dtype)
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_replace.py:213]: can you put the issue number as a comment here? see some other tests for an example

File: pandas/tests/series/methods/test_replace.py [hunk 2]
```diff
@@ -208,6 +208,14 @@ def test_replace_with_dict_with_bool_keys(self):
...
+        result.replace(0, pd.NA)
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_replace.py:214]: I think you'll want
```
result = result.replace(0, pd.NA)
```

File: pandas/tests/series/methods/test_replace.py [hunk 3]
```diff
@@ -208,6 +208,14 @@ def test_replace_with_dict_with_bool_keys(self):
...
+        expected = pd.Series([0, None]).astype(dtype)
+        expected.fillna(0).replace(0, pd.NA)
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_replace.py:216]: why is this line needed? Just construct `expected` directly

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @ftrihardjo

**Comment by ftrihardjo** [comment]
thank you for your review

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

File: pandas/tests/series/methods/test_replace.py [hunk 4]
```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["int8", "int16", "int32", "int64"])
+    def test_replace_int_with_na(self, dtype):
+        # GH 38267
+        result = pd.Series([0, None]).astype(dtype).replace(0, pd.NA)
+        expected = pd.Series([0, None]).astype(dtype).fillna(0).replace(0, pd.NA)
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_replace.py:215]: Is it possible to construct `expected` directly (e.g. `expected = pd.Series([pd.NA, pd.NA])`)?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/tests/series/methods/test_replace.py [hunk 5]
```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["int8", "int16", "int32", "int64"])
```
Comment (MarcoGorelli) [inline pandas/tests/series/methods/test_replace.py:211]: the types need to be capitalised, as in the issue, e.g. `Int64` not `int64`

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/tests/series/methods/test_replace.py [hunk 6]
```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["Int8", "Int16", "Int32", "Int64"])
+    def test_replace_int_with_na(self, dtype):
```
Comment (arw2019) [inline pandas/tests/series/methods/test_replace.py:212]: I'd call it `test_replace_nullable_int_with_na` or `test_replace_Int_with_na`

File: pandas/tests/series/methods/test_replace.py [hunk 7]
```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
...
+        # GH 38267
+        result = pd.Series([0, None]).astype(dtype).replace(0, pd.NA)
+        expected = pd.Series([pd.NA, pd.NA])
```
Comment (arw2019) [inline pandas/tests/series/methods/test_replace.py:215]: you also need to pass dtype here else (looking at CI output) this casts to object. That's why currently this test doesn't actually pass

**Comment by MarcoGorelli** [comment]
Thanks @ftrihardjo , looks like just https://github.com/pandas-dev/pandas/pull/38693#discussion_r549074576 still needs to be addressed

File: pandas/tests/series/methods/test_replace.py [hunk 8]
```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["Int8", "Int16", "Int32", "Int64"])
+    def test_replace_Int_with_na(self, dtype):
+        # GH 38267
+        result = pd.Series([0, None], dtype=dtype).replace(0, pd.NA)
```
Comment (jreback) [inline pandas/tests/series/methods/test_replace.py:214]: can you add the working example as well e.g. 0,1 as another case.

File: pandas/tests/series/methods/test_replace.py [hunk 9]
```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["Int8", "Int16", "Int32", "Int64"])
```
Comment (jreback) [inline pandas/tests/series/methods/test_replace.py:211]: can you use the fixture: any_nullable_int_dtype instead here.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ftrihardjo

**Pull request merged**